src/Sia.Data.Incident/Filters/EventFilters.cs

```diff
@@ -34,7 +36,12 @@ public override IQueryable<Event> Filter(IQueryable<Event> source)
                 var workingCompare = String.IsNullOrEmpty(DataValue)
                     ? String.Format(KeyComparison, DataKey)
                     : String.Format(KeyValueComparison, new string[] { DataKey, DataValue });
-                working = working.Where(obj => obj.Data.Contains(workingCompare));
+                working = working.Where(obj => ((obj.Data == null) || obj.Data.Contains(workingCompare)));
+            }
+
+            if (!String.IsNullOrEmpty(DataSearch))
+            {
+                working = working.Where(obj => ((obj.Data == null) || obj.Data.Contains(DataSearch)));
```

Comment:
We should not match on null data, since an event without data definitely does not contain the phrase we're looking for.

obj.Data != null && obj.Data.Contains(DataSearch)

After the merge from master (which changes the method from Filter to IsMatchFor), this will probably need to look like:

if (!string.IsNullOrEmpty(DataSearch) && (toCompare.Data == null || !toCompare.Data.Contains(DataSearch)) { return false; }